pymc_marketing/customer_choice/nested_logit.py

```diff
@@ -0,0 +1,783 @@
+#   Copyright 2022 - 2025 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Nested Logit for Product Preference Analysis."""
+
+import json
+
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import patsy
+import pymc as pm
+import pytensor.tensor as pt
+from typing_extensions import Self
+
+from pymc_marketing.model_builder import ModelBuilder
+from pymc_marketing.model_config import parse_model_config
+from pymc_marketing.prior import Prior
+
+HDI_ALPHA = 0.5
+
+
+class NestedLogit(ModelBuilder):
+    """
+    Nested Logit class.
```

Comment:
Could we add some references to the doctrings? (books or papers?)

Reply:
Done